packages/bpk-component-modal/src/BpkModal.js

```diff
@@ -123,7 +123,7 @@ export const defaultProps = {
   closeOnEscPressed: true,
 };
 
-BpkModal.propTypes = { ...propTypes };
-BpkModal.defaultProps = { ...defaultProps };
+BpkModal.propTypes = { ...propTypes, onClose: PropTypes.func };
```

Comment:
Rather than adding here you can probably move this to as looking at this it would be defining the onClose twice:

https://github.com/Skyscanner/backpack/pull/1937/files#diff-9b7c4907d7e12003dbda71f207de17f5R106

Reply:
I initially had - but chose to instead overwrite it here.

The spread propTypes and defaultProps come from BpkModalDialog, where they are defined as:
```
onClose: onClosePropType,
```
and
```
onClose: null,
```
https://github.com/Skyscanner/backpack/blob/master/packages/bpk-component-modal/src/BpkModalDialog.js#L140

This is correct afaik for `BpkModalDialog` - it correctly _can_ take null or a function, i.e the Flow type is:
```
onClose: ?(event: SyntheticEvent<>) => void,
```

Because of this I don't want to change anything for `BpkModalDialog`, but as its props are all otherwise exactly the same continue to import and spread them here as before, and override the one that is now different for `BpkModal`.

Does that make sense? Separately, is that correct? 😂